volumes/csi-wrapper/README.md

```diff
@@ -0,0 +1,302 @@
+```
+> **Note** The `ibm-vpc-block-csi-node-r7b5w` pod won't in `Running` status as it's on control node, just ignore it.
+
+7. Create **storage class** for Peerpod:
+```bash
+kubectl apply -f hack/ibm/ibm-vpc-block-5iopsTier-StorageClass-for-peerpod.yaml
+```
+
+## Run the `csi-wrapper for peerpod storage` demo
+
+1. Create one pvc that use `ibm-vpc-block-csi-driver`
+```bash
+kubectl create -f hack/ibm/my-pvc-kube-system.yaml
+```
+
+2. Wait for the pvc status to become `bound`
+```bash
+kubectl -n kube-system get pvc
+NAME     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                AGE
+my-pvc   Bound    pvc-8e258388-3493-45fd-a92b-e4576ccdce7c   10Gi       RWO            ibmc-vpc-block-5iops-tier   19s
+```
+
+3. Create nginx peer-pod demo with with `podvm-wrapper` and `ibm-vpc-block-csi-driver` container
+```bash
+kubectl create -f hack/ibm/nginx-kata-with-my-pvc-and-csi-wrapper.yaml
+```
+
+4. Wait 2 minutes, check if the `nginx` pod is running:
+```bash
+kubectl get po -A | grep nginx
+kube-system   nginx                                                 3/3     Running             0              101s
+```
+
+5. Check the `PeerpodVolume` crd object and the state is `nodePublishVolumeApplied`
+- Getting `PeerpodVolume` objects:
+```bash
+kubectl get PeerpodVolume -A
+NAMESPACE     NAME                                        AGE
+kube-system   r022-31f30b9b-556b-4a5d-80b0-419270a8783e   11m
+```
+- Using name get the state of the PeerpodVolume:
+```bash
+kubectl -n kube-system get PeerpodVolume r022-31f30b9b-556b-4a5d-80b0-419270a8783e -o yaml |grep state
+  state: nodePublishVolumeApplied
+```
+
+6. Exec into nginx container check the mount:
+```bash
+kubectl exec nginx -n kube-system -c nginx -i -t -- sh
+# lsblk
+NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
+loop0     7:0    0 91.8M  1 loop
+loop1     7:1    0 63.2M  1 loop
+loop2     7:2    0 49.6M  1 loop
+vda     252:0    0  100G  0 disk
+|-vda1  252:1    0 99.9G  0 part /run/secrets/kubernetes.io/serviceaccount
+|-vda14 252:14   0    4M  0 part
+`-vda15 252:15   0  106M  0 part
+vdb     252:16   0  372K  0 disk
+vdc     252:32   0   44K  0 disk
+vdd     252:48   0   10G  0 disk /mount-path
+# ls /mount-path
+lost+found
+```
+> **Note** We can see, the device **vdd** is mounted to `/mount-path` in nginx container as expected.
+
+- Create a file under `/mount-path`
+```bash
+echo "from nignx container date:" $(date) > /mount-path/incontainer.log
+```
+- On the development machine check the created vsi pod:
+```bash
+ibmcloud is ins |grep nginx
+02e7_d9d5d4f4-d188-4c96-8f7b-c390b97f7aa4   podvm-nginx-1c0b9c12    running   10.244.0.4     -                 bx2-2x8    podvm-82ae9f8-dirty-amd64            liudali-csi-wrapper-vpc   jp-tok-1   default
+```
+- Using the VSI pod ID, get the network interface id:
+```bash
+ibmcloud is in 02e7_d9d5d4f4-d188-4c96-8f7b-c390b97f7aa4 | grep -A 2  "Network Interfaces"
```

Comment:
If we decide that jq is a bit more friendly this could become:
```
# ibmcloud is in 02e7_d9d5d4f4-d188-4c96-8f7b-c390b97f7aa4 --output json | jq -r '.network_interfaces[].id'
02e7-663133fd-94aa-4d6b-bf2f-2163280dabf1
```